cvat-ui/src/actions/export-actions.ts

```diff
@@ -75,52 +78,33 @@ export const exportActions = {
     ),
 };
 
-export async function listenExportDatasetAsync(
-    rqID: string,
-    dispatch: (action: ExportActions | RequestsActions) => void,
-    params: {
-        instance: ProjectOrTaskOrJob | RequestInstanceType,
-        format: string,
-        saveImages: boolean,
-    },
-): Promise<void> {
-    const { instance, format, saveImages } = params;
-    const resource = saveImages ? 'dataset' : 'annotations';
-
-    const instanceType = getInstanceType(instance);
-    try {
-        const result = await listen(rqID, dispatch);
-        const target = !result?.url ? 'cloudstorage' : 'local';
-        dispatch(exportActions.exportDatasetSuccess(
-            instance, instanceType, format, resource, target,
-        ));
-    } catch (error) {
-        dispatch(exportActions.exportDatasetFailed(instance, instanceType, format, resource, error));
-    }
-}
-
+/** *
+ * Function is supposed to be used when a new dataset export request initiated by a user
+** */
 export const exportDatasetAsync = (
     instance: ProjectOrTaskOrJob,
     format: string,
     saveImages: boolean,
     useDefaultSettings: boolean,
     targetStorage: Storage,
     name?: string,
-): ThunkAction => async (dispatch, getState) => {
-    const state = getState();
-
+): ThunkAction => async (dispatch) => {
     const resource = saveImages ? 'dataset' : 'annotations';
     const instanceType = getInstanceType(instance);
 
     try {
         const rqID = await instance.annotations
             .exportDataset(format, saveImages, useDefaultSettings, targetStorage, name);
-        if (shouldListenForProgress(rqID, state.requests)) {
-            await listenExportDatasetAsync(rqID, dispatch, {
-                instance, format, saveImages,
+
+        if (rqID) {
+            await core.requests.listen(rqID, {
+                callback: (updatedRequest) => updateRequestProgress(updatedRequest, dispatch),
             });
-        }
-        if (!rqID) {
+            const target = targetStorage.location === StorageLocation.LOCAL ? 'local' : 'cloudstorage';
```

Comment:
Would it be possible to use the StorageLocation enum as the target type here instead of string literals? It seems to have the same values with slight differences (e.g., local, cloud_storage). I assume these literals are primarily used to display the correct message.

Reply:
We can fix it all over this file